### PR TITLE
MD5 signature validation using ordinal compare + UrlDecode

### PR DESCRIPTION
**Pull request by janpieterz**
Fixes part of #140

### CODE REVIEW

**Comment by janpieterz** [comment]
Happy to contribute other hashing methods as well but this'd need a bigger refactor as I reckon we should unit test that.

**Comment by janpieterz** [comment]
Refactor suggestions: 

1. Have enum with hashing options (MD5HASH, HMACSHA512 etc etc)
1. Validate time as first thing in this method, then have internal static bool method CalculateAndValidateSignature
1. Unit test this internal method. We'll need some sample signed items from Nexmo's backend for this. Maybe there's a Nexmo test number that can be used and someone can generate the requests for all of those?

**Comment by judy2k** [comment]
Thank you so much for this contribution!

I have somehow become our expert on our hashing algorithms, so tomorrow I'll take this on and add the other hashing algorithms to it and get them tested.

File: Nexmo.Api/Request/SignatureHelper.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public static bool IsSignatureValid(IEnumerable<KeyValuePair<string, StringValue
                 return false;
             }
 
-            var sorted = new SortedDictionary<string, string>();
+            var sorted = new SortedDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```
Comment (judy2k) [inline Nexmo.Api/Request/SignatureHelper.cs]: I think the keys are all lower case, so this shouldn't make a difference?

Comment (janpieterz) [reply Nexmo.Api/Request/SignatureHelper.cs]: Shouldn't indeed.

**Comment by slorello89** [comment]
@janpieterz thank you for this - I wish I had seen the PR earlier because I ran into the same ordinal sorting issue when I was implementing the validation! I actually implemented all the signing / validation for the SDK in a different PR so I'm going to close this one - all the same thanks so much! 

For more details about how to use the SDK for signing / validation please see: https://developer.nexmo.com/concepts/guides/signing-messages

**Pull request closed without merging**